plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+MAX_LOOKBACK_DAYS = 7
+DEFAULT_LOOKBACK_HOURS = 24
+ALERT_LIMIT = 100
+
+MAX_LIMIT = 7500
+
+# State held values
+LAST_ALERT_TIME = "last_incident_time"
+LAST_ALERT_HASH = "last_incident_hash"
+# State held values = CONOR
+LAST_SEARCH_FROM = "last_search_from"
+LAST_SEARCH_TO = "last_search_to"
+
+# Used to paginate
+ALERTS_OFFSET = "alerts_offset"
+FROM_TIME_FILTER = "from_time_filter"
+TO_TIME_FILTER = "to_time_filter"
+TIMESTAMP_KEY = "event_timestamp"
+
+NEXT_PAGE_LINK = "next_page_link"
+# Custom imports below
+
+
+class MonitorAlerts(insightconnect_plugin_runtime.Task):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_alerts",
+            description=Component.DESCRIPTION,
+            input=MonitorAlertsInput(),
+            output=MonitorAlertsOutput(),
+            state=MonitorAlertsState(),
+        )
+
+    def run(self, params={}, state={}, custom_config: dict = {}):  # pylint: disable=unused-argument
+        existing_state = state.copy()
+        has_more_pages = False
+        parameters = {}
+        print(f"{existing_state = }")
+
+        custom_config = {}
+        state = {LAST_SEARCH_FROM: None, LAST_SEARCH_TO: None}
+
+        try:
+
+            now_time = self._get_current_time()
+            now = now_time - timedelta(minutes=15)
+            # last 15 minutes
+            end_time = now.strftime(TIME_FORMAT)
+            print(f"{end_time = }")
+            start_time, alert_limit = self._parse_custom_config(custom_config, now, state)
+
+            self.logger.info("Starting to download alerts...")
+
+            response, state, has_more_pages = self.get_alerts_palo_alto(state=state, custom_config=custom_config)
+
+            # TODO - Are we supposed to show the total (MAX LIMIT) or the total (total_count)
+            self.logger.info(f"Total alerts returned: {len(response)}")
+
+            # Local Debugging
+            self.logger.info(f"{type(response) = }")
+            self.logger.info(f"{len(response) = }")
+            self.logger.info(f"{state}, {has_more_pages = }")
+
+            # TODO - Change first None to response
+            # It's set to none so we can see all the logging without the results getting in the way
+
+            # Note - when running locally, we should only get the first 100
+            # When has more pages is True, the task will be rerun automatically again (on staging / the cloud exec)
+
+            # If has more pages == True, then we need to change the search index from 0-100 to 101-200 (Done below now I think)
+            return response, state, has_more_pages, 200, None
+
+        except PluginException as error:
+            self.logger.error(
+                f"A PluginException has occurred. Status code 500 returned. Error: {error.cause}. "
+                f"Existing state: {existing_state}"
+            )
+            return [], existing_state, False, 500, PluginException(cause=error.cause, data=error)
```

Comment:
It doesn't seem like 'failed' status codes are being correctly bubbled up (`500` is hardcoded)

e.g. in the case of invalid credentials, we'd want to return a `401` here, so that the integration goes into a FATAL state